src/FileFormats/format_adlibgold_bnk2.cpp

```diff
@@ -0,0 +1,258 @@
+/*
+ * OPL Bank Editor by Wohlstand, a free tool for music bank editing
+ * Copyright (c) 2018 Vitaly Novichkov <admin@wohlnet.ru>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "format_adlibgold_bnk2.h"
+#include "../common.h"
+#include <QDebug>
+#include <cstring>
+
+static const char *AdLibGoldBnk2_magic = "Accomp. Bank, (C) AdLib Inc";
+
+bool AdLibGoldBnk2_reader::detect(const QString &, char *magic)
+{
+    return !std::memcmp(magic, AdLibGoldBnk2_magic, 28);
+}
+
+static void convertInstrument(
+    const uint8_t src[28], FmBank::Instrument &dst, const char *name)
+{
+    unsigned char C4xxxFFFC = src[24];
+    unsigned char xxP24NNN = src[25];
+    unsigned char TTTTTTTT = src[26];
+    // unsigned char xxxxxxxx = src[27];
+
+    dst.is_blank = false;
+    std::strcpy(dst.name, name);
+
+    for(unsigned nth_op = 0; nth_op < 4; ++nth_op)
+    {
+        static const unsigned opnums[4] =
+            { MODULATOR1, CARRIER1, MODULATOR2, CARRIER2 };
+        unsigned opnum = opnums[nth_op];
+
+        const uint8_t *op = src + 6 * nth_op;
+        dst.setAVEKM(opnum, op[0]);
+        dst.setKSLL(opnum, op[1]);
+        dst.setAtDec(opnum, op[2]);
+        dst.setSusRel(opnum, op[3]);
+        dst.setWaveForm(opnum, op[4]);
+    }
+    dst.setFBConn1(C4xxxFFFC & 0x0F);
+    dst.setFBConn2((C4xxxFFFC & 0x0E) | (C4xxxFFFC >> 7));
+
+    dst.fine_tune = (int8_t)TTTTTTTT;
+    dst.en_4op = (xxP24NNN & 8) != 0;
+    dst.en_pseudo4op = dst.en_4op && (xxP24NNN & 16) != 0;  // pseudo-4op maybe?
+
+    //--- rhythm mode percussion: BD=0, SD=1, TOM=2, CYMB=3, HH=4
```

Comment:
Instruments structure has the field where you can set percussion instrument as rhythm-mode. Why not to try to also import that?

Reply:
Because I had some doubt on interpreting the fields, but I speculate that's how it would go:

If the bit `is_perc` is set, then consider rhythm mode and use it.

As I observed from Master of Magic banks, none of drums had the percussion bit set in them. I'm so guessing this must be the purpose.

Have we other examples of BNK2?